File: clojure/day2/unless.clj
(ns unless
  (:gen-class))
(println "----------------")

;; (defn unless [condition, body] (if (not condition) body, nil))

;; (defn unless [condition & body] (if (not condition) (do body)))

;; (defmacro unless [condition & body] `(if (not ~condition) (do ~@body)))

(defmacro unless [test body] (list 'if (list 'not test) body))

(println (unless true (println "This shouldn't print")))
(println (unless false (println "This should print")))

File: julia/day3/unless.jl
println("-------------------")

macro unless(condition, body)
    return quote
        if !($condition)
            $body
        end
    end
end

a = [1, 2, 3, 4, 5]

@unless isempty(a) println("a is not empty")

@unless in(a, 6) println("6 is not in a")
